plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/helpers.ts

```diff
@@ -351,3 +355,22 @@ function extractCollaboratorName(
   if ('user' in collaborator) return collaborator.user;
   return collaborator.team;
 }
+
+async function validateAccessTeam(client: Octokit, access: string) {
+  const [org, team_slug] = access.split('/');
+  try {
+    // Below rule disabled because of a 'getByName' check for a different library
+    // incorrectly triggers here.
+    // eslint-disable-next-line testing-library/no-await-sync-query
+    await client.rest.teams.getByName({
+      org,
+      team_slug,
+    });
```

Comment:
Just to see if we can be a bit more efficient here, is there no way to determine that this was the reason that the repo creation failed? If not, I think we could perhaps flip this around so that this request is only made in the event that repo creation fails to highlight the error. That way we avoid an extra request for the happy path.

Reply:
Unfortunately, the problem is that the repo creation itself succeeds in the original implementation; the problem occurs when after the repo is created and the permissions fail to apply. An error kicks you out of the task, and you end up with a bare repo with no contents and no permissions applied. 

This ends up causing headache in scenarios where the person running the task doesn't have permissions to clean up GitHub; they created a bare repo, subsequent runs tell you the repo name is taken, and they need to contact someone else to perform cleanup.